cmd/livepeer/livepeer.go

```diff
@@ -183,17 +191,17 @@ func main() {
 	} else if transcoder {
 		n.NodeType = core.TranscoderNode
 		glog.Info("***Livepeer is in transcoder mode ***")
-		server.RunTranscoder(n, *orchAddr)
+		server.RunTranscoder(n, orchAddresses[0])
```

Comment:
We probably need a length check here

Reply:
Since we're checking for `transcoder` on line 191, and transcoder can only be true if [`if len(*orchAddr) > 0`](https://github.com/livepeer/go-livepeer/blob/dd5dc14ed2976655b299c2c9f7fffbe9c1e0f365/cmd/livepeer/livepeer.go#L103) (I'll change this line to check for the len of the array), then is there a danger to not checking len here?

Reply:
Good point. One has to be careful to follow the logic -- `transcoder` can be set with either the `-transcoder` flag or the `-standaloneTranscoder` flag, and the `-transcoder` flag results in line 189 being checked, whereas `-standaloneTranscoder` can only enable `transcoder`  mode if `orchAddr` is non-nil.

Overall it's a bit fragile and we probably should refactor this soon for robustness. The good thing is, if this flow ever breaks, it'll likely crash at start-up time, making the issue easier to diagnose.

Reply:
Ah yes, we'll have to refactor soon! I'll do a len check here until we can figure out a better system for these flags. Meanwhile, that's right, an immediate-crash is easy to diagnose!